src/components/Application.tsx

```diff
@@ -85,6 +85,9 @@ interface State {
   productLines?: Collection<beachfront.ProductLine>
 
   // Map state
+  mapMode?: string
+  detections?: (beachfront.Job | beachfront.ProductLine)[]
```

Comment:
What's the overall reason for adding the detections and frames as state properties? Is there any downsides to this?
Was this required by the meridian thing or just a nice refactor anyways?